wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/WireJsonAdapterFactory.kt

```diff
@@ -82,18 +82,28 @@ class WireJsonAdapterFactory private constructor(
     if (nextAnnotations != null) {
       return when (type) {
         Boolean::class.javaObjectType,
-        Boolean::class.javaPrimitiveType -> moshi.adapter<Boolean>(type).omitValue(false)
+        Boolean::class.javaPrimitiveType -> {
+          moshi.adapter<Boolean>(type, nextAnnotations).omitValue(false)
```

Comment:
RAD.

Reply:
Also I take back everything I said about omitValue() in the previous PR. Now that this spans types this design makes total sense